.github/workflows/update-wpt.yml

```diff
@@ -0,0 +1,67 @@
+name: WPT update
+
+on:
+  schedule:
+    # Run once a week at 12:00 AM UTC on Sunday.
+    - cron: 0 0 * * *
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  wpt-subsystem-update:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        subsystem:
+          - url
+
+    steps:
+      - uses: actions/checkout@v4@<hash>
+        with:
+          persist-credentials: false
+
+      - name: Set up GitHub Authentication
+        run: echo "GitHub Authentication set up"
+        env:
+          GH_USER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Update WPT for subsystem ${{ matrix.subsystem }}
+        run: |
+          git node wpt update --subsystem=${{ matrix.subsystem }}
+        env:
+          GH_USER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Run WPT tests
+        run: |
+          git node wpt run --subsystem=${{ matrix.subsystem }}
+        env:
+          GH_USER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Handle test failures and success
+        run: |
+          if [[ "$(git status --porcelain)" != "" ]]; then
+            git config --global user.name "${{ github.actor }}[bot]"
+            git config --global user.email "${{ github.actor }}@users.noreply.github.com"
+            git checkout -b actions/update-wpt-${{ matrix.subsystem }}
+            git add .
+            git commit -m "test: update WPT for ${{ matrix.subsystem }} to ${{ env.NEW_VERSION }}"
+            gh auth setup-git
+            git push origin actions/update-wpt-${{ matrix.subsystem }} -fu --no-verify
+            echo "createPR=true" >> "$GITHUB_OUTPUT"
+          else
+            echo "No changes to commit."
+          fi
+        env:
+          GH_USER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Open a PR for the subsystem update
+        if: steps.push-changes.outputs.createPR
+        continue-on-error: true
+        run: |
+          gh pr create --head actions/update-wpt-${{ matrix.subsystem }} \
+          --title "test: update WPT for ${{ matrix.subsystem }} to ${{ env.NEW_VERSION }}" \
+          --body "This is an automated update of the WPT for ${{ matrix.subsystem }} to ${{ env.NEW_VERSION }}."
+        env:
+          GH_USER_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
WDYT about the create-or-update-pr action?

Reply:
I think it is healthier to have an update

Reply:
I thought of this to indicate the purpose of the mr that will occur here, since this is an update, wouldn't it be good to update it, thank you for your comments, I will apply it, I'm just curious 🙏❤️🚀

Reply:
I applied here  🙏